File: Answer/3517.最小回文排列-i.go
/*
 * @lc app=leetcode.cn id=3517 lang=golang
 *
 * [3517] 最小回文排列 I
 */

// @lc code=start
func smallestPalindrome(s string) string {
	n := len(s)
	cnt := [26]int{}
	for i := 0; i < n/2; i++ {
		cnt[s[i]-'a']++
	}
	ans := make([]byte, n)
	left, right := 0, n-1
	for i := 0; i < 26; i++ {
		c := byte('a' + i)
		for cnt[i] > 0 {
			ans[left] = c
			ans[right] = c
			left++
			right--
			cnt[i]--
		}
	}
	if n%2 == 1 {
		ans[left] = s[n/2]
	}
	return string(ans)
}

// @lc code=end

File: Answer/3517.最小回文排列-i.py
#
# @lc app=leetcode.cn id=3517 lang=python3
#
# [3517] 最小回文排列 I
#

# @lc code=start
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        n = len(s)
        cnt = [0] * 26
        for ch in s[:n // 2]:
            cnt[ord(ch) - ord('a')] += 1
        ans = [''] * n
        left, right = 0, n - 1
        for i in range(26):
            c = chr(ord('a') + i)
            while cnt[i] > 0:
                ans[left] = c
                ans[right] = c
                left += 1
                right -= 1
                cnt[i] -= 1
        if n % 2 == 1:
            ans[left] = s[n // 2]
        return ''.join(ans)
# @lc code=end
